File: gencode/java/udmi/schema/Level.java
package udmi.schema;

import java.util.HashMap;
import java.util.Map;

// This class is manually curated and then copied into the gencode directory. Look for the
// proper source and don't be fooled! This is subset of the StackDriver LogSeverity levels.
public enum Level {

  DEBUG(100),
  INFO(200),
  NOTICE(300),
  WARNING(400),
  ERROR(500);

  private final int value;
  private final static Map<Integer, Level> CONSTANTS = new HashMap<>();

  static {
    for (Level c: values()) {
      CONSTANTS.put(c.value, c);
    }
  }

  Level(int value) {
    this.value = value;
  }

  public int value() {
    return this.value;
  }

  public static Level fromValue(int value) {
    Level constant = CONSTANTS.get(value);
    if (constant == null) {
      throw new IllegalArgumentException(Integer.toString(value));
    } else {
      return constant;
    }
  }
}

File: gencode/python/udmi/schema/level.py
"""
This module defines the Level enum, representing logging severity levels
aligned with a subset of StackDriver LogSeverity levels.
"""
from enum import IntEnum, unique

@unique
class Level(IntEnum):
  """
  This class is manually curated. This is a subset of the
  StackDriver LogSeverity levels.
  """
  INVALID = -1
  TRACE = 50
  DEBUG = 100
  INFO = 200
  NOTICE = 300
  WARNING = 400
  ERROR = 500
  # Note: 'CRITIAL' typo preserved from Java source
  CRITIAL = 600

  @classmethod
  def from_value(cls, value: int):
    """
    Look up the Enum member by its integer value.
    """
    try:
      return cls(value)
    except ValueError as exc:
      raise ValueError(str(value)) from exc
